test/mpi/test_mpi_cuda.c

```diff
@@ -0,0 +1,668 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucs/debug/log_def.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <mpi.h>
+#include <cuda.h>
+#include <pthread.h>
+
+#define SIZE_S 8
+#define SIZE_M 4 * 1024 * 1024
+#define SIZE_L 16 * 1024 * 1024
+
+#define TEST_NAME_FMT      "TEST[%d:%-17s:%-17s:*       ]"
+#define TEST_NAME_SIZE_FMT "TEST[%d:%-17s:%-17s:%-8zi]"
+
+#define PRINT_ROOT(fmt, ...) \
+    do { \
+        int _rank; \
+        MPI_Comm_rank(MPI_COMM_WORLD, &_rank); \
+        if (_rank == 0) { \
+            fprintf(stdout, fmt, ##__VA_ARGS__); \
+        } \
+    } while (0)
+
+#define array_size(_array) (sizeof(_array) / sizeof((_array)[0]))
+
+#define CUDA_CALL(_func) \
+    do { \
+        CUresult __err = (_func); \
+        if (__err != CUDA_SUCCESS) { \
+            const char *__err_str; \
+            cuGetErrorString(__err, &__err_str); \
+            fprintf(stderr, "test_mpi_cuda.c:%-3u %s failed: %d (%s)\n", \
```

Comment:
can you use `__FILE__` instead?

Reply:
It will print full path which is not needed here. And I don't want to add code to fetch a filename from the __FILE__, since it overcomplicates simple code.

Reply:
It will print full path which is not needed here. And I don't want to add code to fetch a filename from the FILE, since it overcomplicates simple code.

Reply:
will it be just something like below?
```
const char *filename = strrchr(__FILE__, '/');
filename = filename ? filename + 1 : __FILE__;
```
imo, it is better, so no need to change the code if file is renamed for some reason. Another alternative is just do not print a file name, because it is anyway a single file test

Reply:
Removed.